torch/distributed/launch.py

```diff
@@ -76,18 +93,24 @@
 3. In your training program, you are supposed to call the following function
 at the beginning to start the distributed backend. You need to make sure that
 the init_method uses ``env://``, which is the only supported ``init_method``
-by this module:
+by this module.
+
+::
 
-    ``torch.distributed.init_process_group(backend='YOUR BACKEND',
-    init_method='env://')``
+    torch.distributed.init_process_group(backend='YOUR BACKEND',
+                                         init_method='env://')
 
 4. In your training program, you can either use regular distributed functions
-or use DistributedDataParallel module. If your training program uses GPUs for
-training and you would like to use DistributedDataParallel module, here is how
-to configure it.
+or use :func:`torch.nn.parallel.DistributedDataParallel` module. If your
+training program uses GPUs for training and you would like to use
+:func:`torch.nn.parallel.DistributedDataParallel` module,
+here is how to configure it.
 
-    ``model = torch.nn.parallel.DistributedDataParallel(model,
-    device_ids=[arg.local_rank], output_device=arg.local_rank)``
+::
+
+    model = torch.nn.parallel.DistributedDataParallel(model,
+                                                      device_ids=[arg.local_rank],
```

Comment:
I do not understand this, I think `len(device_ids)` should be `nproc_per_node` so that the DistributedDataParallel will use `nproc_per_node`'s GPUs in the node, but it seems that DistributedDataParallel use only one device which is `arg.local_rank` here.

Reply:
In this setting (which I think is the fastest), we launch as many processes as there are GPUs in the machine, and each process will only use one GPU. Does this answer your question?

Reply:
@fmassa Yes, I find this too, thank you.